src/sound_chatot.c

```diff
@@ -0,0 +1,187 @@
+#include "sound_chatot.h"
+#include "sound.h"
+#include "constants/species.h"
+#include "math_util.h"
+#include "unk_02004A44.h"
+#include <nitro/spi/ARM9/mic.h>
+
+BOOL ChatotSoundMain(void) {
+    u8 *r4 = GF_SdatGetAttrPtr(16);
+    u8 *r0 = GF_SdatGetAttrPtr(30);
+
+    if (*r0 == 1) {
+        if (*r4 == 1) {
+            if (sub_02005738(14) == 0) {
+                sub_02006DB8();
+                return TRUE;
+            }
+        }
+        else {
+            sub_02006DB8();
+            return TRUE;
+        }
+    }
+
+    return FALSE;
+}
+
+BOOL Chatot_checkCry(SOUND_CHATOT *a0) {
+    u8 *r5 = GF_SdatGetAttrPtr(31);
+    u8 *r4 = GF_SdatGetAttrPtr(54);
+
+    if (!Chatot_exists(a0)) {
+        return FALSE;
+    }
+
+    if (*r4 == 1) {
+        return FALSE;
+    }
+
+    if (*r5 != 1) {
+        return TRUE;
+    }
+
+    return FALSE;
+}
+
+BOOL sub_02006D04(SOUND_CHATOT *a0, u32 a1, s32 a2, s32 a3) {
+    s8 *sp0 = sub_020059D8();
+    s8 *sp4 = GF_SdatGetAttrPtr(30);
+
+    if (!Chatot_checkCry(a0)) {
+        return FALSE;
+    }
+
+    sub_02006300(0);
+    sub_02006DB8();
+    sub_02005600(14);
+    u16 r4 = (LCRandom() % 8192);
+
+    Chatot_Decode(sp0, Chatot_GetData(a0));
+
+    UnkStruct_02004A44_0 sp8;
+
+    sp8.unk00 = sub_020055AC(14);
+    sp8.unk04 = 0;
+    sp8.unk08 = sub_020059D8();
+    sp8.unk0c = 0;
+    sp8.unk10 = 0;
+    sp8.unk14 = 2000;
+    sp8.unk18 = 2000;
+    sp8.unk20 = (u32)(r4 + 0x8000);
+    sp8.unk24 = a3 / 2 + 64;
+    sp8.unk1c = a2;
+
+    BOOL ret = sub_020056E8(&sp8, 14);
+    sub_02005774(14, a2);
+    *sp4 = 1;
+    sub_02006E3C(0);
+
+    return ret;
+}
+
+void sub_02006DB8() {
+    u8 *r5 = GF_SdatGetAttrPtr(16);
+    u8 *r4 = GF_SdatGetAttrPtr(30);
+
+    if (*r5 == 1) {
+        sub_020058B8(14);
+        sub_02005680(14);
+    }
+
+    *r4 = 0;
+}
+
+u32 Chatot_startRecording(void) {
+    MIC_SamplingData st0;
```

Comment:
Let's copy the `MICAutoParam` data type from NitroSDK and use that instead.

Reply:
Should I copy over the enum names and struct field names directly as they are in the SDK?

Reply:
Yes. Be sure to reformat it to be consistent with the style of the rest of the repo too. The definitions go in <lib/include/nitro/spi/mic.h>